Converters/Tekla/Speckle.Converters.TeklaShared/ToSpeckle/Helpers/UserDefinedAttributesExtractor.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+
+namespace Speckle.Converters.TeklaShared.ToSpeckle.Helpers;
+
+public class UserDefinedAttributesExtractor
+{
+  public Dictionary<string, object?> GetUserDefinedAttributes(TSM.ModelObject modelObject)
+  {
+    var userProperties = new Dictionary<string, object?>();
+    var userValues = new Hashtable();
+
+    if (!modelObject.GetAllUserProperties(ref userValues))
+    {
+      // NOTE: Return empty dictionary if no user properties found or failed to get properties
+      return userProperties;
+    }
+
+    foreach (DictionaryEntry entry in userValues)
+    {
+      if (entry.Value != null && !string.IsNullOrEmpty(entry.Value.ToString()))
+      {
+        var propertyName = entry.Key.ToString();
+        var propertyValue = entry.Value;
+
+        if (propertyName != null)
+        {
+          var reportProperty = new Dictionary<string, object?> { ["name"] = propertyName, ["value"] = propertyValue };
```

Comment:
these should be entries in the `userProperties` dictionary, not single entry dictionaries

Reply:
right, it looks cleaner now. pushed.